backend/gateway/gateway.go

```diff
@@ -303,10 +302,11 @@ func RunWithConfig(f *Flags, cfg *gatewayv1.Config, cf *ComponentFactory, assets
 	}
 
 	srv := &http.Server{
-		Handler:      mux.InsecureHandler(rpcMux),
-		Addr:         addr,
-		ReadTimeout:  timeout,
-		WriteTimeout: timeout,
+		Handler:           mux.InsecureHandler(rpcMux),
+		Addr:              addr,
+		ReadTimeout:       timeout,
+		ReadHeaderTimeout: timeout,
```

Comment:
I don't think you need to set `ReadHeaderTimeout`:
```
If ReadHeaderTimeout is zero, the value of ReadTimeout is used. If both are zero, there is no timeout.
```

Reply:
oh yeah let me revert this, this was from a warning in Go 1.17 only I think